defender/wallet-migrator/autotask-1/index.js

```diff
@@ -0,0 +1,173 @@
+const stackName = 'wallet_migrator';
+const senderWalletAddressSecretName = `${stackName}_SENDER_WALLET_ADDRESS`;
+const receiverWalletAddressSecretName = `${stackName}_RECEIVER_WALLET_ADDRESS`;
+const covalentApiKeySecretName = `${stackName}_COVALENT_API_KEY`;
+const networkSecretName = `${stackName}_NETWORK`;
+
+const axios = require('axios');
+const { ethers } = require('ethers');
+const { DefenderRelayProvider, DefenderRelaySigner } = require('defender-relay-client/lib/ethers');
+
+const erc20Abi = [
+  'function transferFrom(address from, address to, uint256 amount) public returns (bool)',
+  'function allowance(address owner, address spender) external view returns (uint256)',
+];
+
+const erc721Abi = [
+  'function transferFrom(address from, address to, uint256 tokenId) public',
+  'function isApprovedForAll(address owner, address operator) public view returns (bool)',
+];
+
+// returns axios get request to retrieve wallet balance of erc20 and erc721 tokens
+async function get(url, method, headers, auth) {
+  return axios({
+    url,
+    method,
+    headers,
+    auth,
+  });
+}
+
+exports.handler = async function handler(autotaskEvent) {
+  // ensure that the autotaskEvent Object exists
+  if (autotaskEvent === undefined) {
+    throw new Error('autotaskEvent undefined');
+  }
+
+  // ensure that the secrets Object exists
+  const { secrets } = autotaskEvent;
+  if (secrets === undefined) {
+    throw new Error('secrets undefined');
+  }
+
+  const receiverWalletAddress = secrets[receiverWalletAddressSecretName];
+
+  // ensure that the receiverWalletAddress exists
+  if (receiverWalletAddress === undefined) {
+    throw new Error('receiver-wallet-address must be defined in config.dev.yml file or Defender->AutoTask->Secrets');
+  }
+
+  const senderWalletAddress = secrets[senderWalletAddressSecretName];
+
+  // ensure that the senderWalletAddress exists
+  if (senderWalletAddress === undefined) {
+    throw new Error('sender-wallet-address must be defined in config.dev.yml file or Defender->AutoTask->Secrets');
+  }
+
+  const covalentApiKey = secrets[covalentApiKeySecretName];
+
+  // ensure that the covalentApiKey exists
+  if (covalentApiKey === undefined) {
+    throw new Error('covalent-ap-key must be defined in config.dev.yml file or Defender->AutoTask->Secrets');
+  }
+
+  const network = secrets[networkSecretName];
+
+  // ensure that the network exists
+  if (network === undefined) {
+    throw new Error('network must be defined in config.dev.yml file or Defender->AutoTask->Secrets');
+  }
+
+  // get chain id depending on network
+  let chainId;
+  if (network === 'goerli') {
+    chainId = '5';
+  } else if (network === 'mainnet') {
+    chainId = '1';
+  }
```

Comment:
```suggestion
  await signer.getChainId();
```
This can be used after `DefenderRelaySigner` is instantiated.